flink-runtime/src/test/java/org/apache/flink/runtime/operators/DataSinkTaskTest.java

```diff
@@ -276,6 +277,25 @@ public void testSortingDataSinkTask() {
 		}
 	}
 
+	@Test
+	public void testClosingErrorDataSinkTask() throws Exception {
```

Comment:
Though the test does check that `close` was called, it doesn't test if the original issues were resolved (order of `close` and `tryCleanupOnError` on `cancel` and `invoke`).